python/lsst/sims/catUtils/mixins/VariabilityMixin.py

```diff
@@ -238,32 +252,75 @@ def applyAgn(self, params, expmjd_in):
         sfint['z'] = params['agn_sfz']
         sfint['y'] = params['agn_sfy']
         tau = params['agn_tau']
-        epochs = expmjd - toff
-        if epochs.min() < 0:
+
+        # A string made up of this AGNs variability parameters that ought
+        # to uniquely identify it.
+        #
+        cache_name = '%d_%.12f_%.12f_%.12f_%.12f_%.12f_%.12f_%.12f_%.12f' \
+        %(seed, sfint['u'], sfint['g'], sfint['r'], sfint['i'], sfint['z'],
+          sfint['y'], tau, toff)
+
+        resumption = False
+
+        # Check to see if this AGN has already been simulated.
+        # If it has, see if the previously simulated MJD is
+        # earlier than the first requested MJD.  If so,
+        # use that previous simulation as the starting point.
+        #
+        if cache_name in _agn_lc_cache:
+            if _agn_lc_cache[cache_name]['mjd'] <expmjd:
```

Comment:
Good catch.  The method used to be vectorized in expmjd, but that did not make sense in the context of the caching (which I think gets us more than the vectorization, since the vectorization still involved nested for loops).  I was trying to de-vectorize it.  I will revert expmjd to a float.